x-pack/plugin/otel-data/src/main/resources/component-templates/ecs-tsdb@mappings.yaml

```diff
@@ -0,0 +1,12 @@
+version: ${xpack.oteldata.template.version}
+_meta:
+  description: Default mappings for the logs index template installed by x-pack
+  managed: true
+template:
+  mappings:
+    dynamic_templates:
```

Comment:
Let's add a dynamic template that maps boolean fields to keywords. This is to work around https://github.com/elastic/elasticsearch/issues/111338. I think that's preferable over converting booleans to strings in the ES exporter. That's because removing the conversion in the ES exporter would require it to check whether or not the connected ES cluster supports boolean dimensions. When we change it in the dynamic template, the ES exporter doesn't need to know about that and it can always assume that booleans are supported.

cc @carsonip

Reply:
+1. If that's ready, I'll remove the workaround in my draft PR.

QQ: what's the implication on backcompat? If a doc is ingested to 8.16 with the workaround in dynamic template, and 8.17 removed that workaround, will it cause any issues at query time?

Reply:
Not quite sure. Maybe you need to adjust term queries to drop quotes like `"value": "true"` -> `"value": true`. But I'm not too concerned about that as long as this isn't GA. Maybe ES can already coerce this at query time which would make it a complete non-issue then.

Reply:
Pushed https://github.com/elastic/elasticsearch/pull/111091/commits/a9987d38ce15b2fee5674a21974fbdc3da1b5d6b

- Added a test that has `transaction.root: false` that triggers the error `Error extracting routing: Routing values must be strings but found [VALUE_BOOLEAN]`
- Added a dynamic mapping to map boolean to keyword

However that mapping does not work - discussed with @felixbarny, problem is likely that routing happens before dynamic mapping would kick in. Therefore, I commented out that mapping and it will be likely removed from the PR.

For now, the test is also skipped, since the dynamic mapping is not in place.

After slacking with @felixbarny, there is https://github.com/elastic/elasticsearch/pull/111445 open. If that gets in, there won't be a need for `booleans_to_keywords` dynamic mapping.

Alternatively, if https://github.com/elastic/elasticsearch/pull/111445 won't get in, we'll need to look for another approach.

Reply:
As https://github.com/elastic/elasticsearch/pull/111445 is merged, can you rebase (merge main) this PR, so that it is available in a new custom docker build?

Reply:
👍 

We still need `booleans_to_keywords`, I enabled that and also removed skipping the corresponding test which after rebasing on `main` runs through successfully. So it seems to be working and ready to test.